backpack-compose/src/androidTest/kotlin/net/skyscanner/backpack/compose/appsearchmodal/BpkAppSearchModalTest.kt

```diff
@@ -0,0 +1,108 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.appsearchmodal
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.text.buildAnnotatedString
+import junit.framework.Assert.assertFalse
+import kotlinx.coroutines.test.runTest
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.textfield.BpkClearAction
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.City
+import net.skyscanner.backpack.compose.utils.BehaviouralCallback
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+
+class BpkAppSearchModalTest {
+
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    @Test
+    fun withBehaviouralCallbackAndClickActionAdded() = runTest {
+        var didReceiveBehaviouralOnDrawnCallback = false
```

Comment:
I could update these to mocks if needed, wasn't sure if there was a preferred Backpack testing approach but thought I'd keep it simple for the first draft of the test

Reply:
yeah always happy not to use mocks if we don't need to :)